options.go

```diff
@@ -165,7 +170,7 @@ func (opts *options) handleFifos() (io.Writer, error) {
 		}
 
 		if fifo, err = createFifoFileLogs(opts.FcFifoLogFile); err != nil {
-			return nil, fmt.Errorf("failed to create fifo log file: %v", err)
+			return nil, unableToCreateFifoLogFile
```

Comment:
I made an attempt at this but I don't necessarily love how it looks. Appreciate if you have any feedback on it.

Reply:
I would suggest something like this:
```go
return nil, errors.Wrap(err, "failed to create fifo log file");
```

And in test:
```go
errors.Cause(err) == unix.EEXIST
and/or
strings.Contains(err.String(), "failed to create fifo log file")
```

I'm not sure there are other better ways of testing things like these.

Reply:
I don't like the Cause method b/c it requires us to know a lot about what error will happen inside something that is not code for this package and I don't want to care about that.

The strings.Contains I think is mostly what equivalent to what is happening now (I'm using HasPrefix instead of contains). I left the messages as errors just so that the test could print out a somewhat relevant "Expected x but got y " message consistently. Not crazy about this way either b/c it involves checking error strings.

Does this seem crazy?
```go
return nil, MergeError(err, errFailedToCreateLogFifoFile)
...

func MergeError(a, b error) error {
        return mergedError{
                     theirs: a,
                     ours: b,
         }
}

type mergedError{
theirs, ours error
}

func (m *mergedError) Error() string {
return fmt.Sprintf("%s: %s", m.ours.Error(), m.theirs.Error())
}
```
Then in tests it would be like:

```go
if e, ok := err.(*mergedError); ok {
    return e.ours == <our error>
}
```

Reply:
I agree with the sentiment of not wanting to test things outside of this package. Rather than a generic error how about making it an error type more specific to what is being tested.

```go
type FailedFifoLogError struct {
    parent error
}

func (e *FailedFifoLogError) Error() string {
    return fmt.Sprintf("failed to create fifo log file: %v", e.parent)
}
```

Then in the test it would look something more like
```go
if _, ok := err.(*FailedFifoLogError); !ok {
    t.Errorf("error message here")
}
```

What're your thoughts on that? The `MergeError` is a little too generic for my liking and requires inspection of the inner errors to be able to do anything.

Reply:
I don't have super strong feelings about it and am open to doing that something like that if @mxpv and @samuelkarp are good with that approach.

Reply:
@mxpv, @samuelkarp, any opinions?

Reply:
I would prefer simpler approach like `errors.Wrap` + `strings.Contains`, since here you need to create custom type for each error case.

Reply:
So are you happy with the approach used in the PR as it stands now @mxpv?